paimon-web-server/src/main/java/org/apache/paimon/web/server/service/TableService.java

```diff
@@ -0,0 +1,150 @@
+     * @param databaseName The name of the database.
+     * @param tableName The name of the table.
+     * @param key The key of the option to be removed.
+     * @return Returns a {@link R} object indicating the success or failure of the operation. If the
+     *     option is successfully removed, the result will be a successful response with no data. If
+     *     an error occurs during the operation, the result will be a failed response with an error
+     *     code. Possible error codes: {@link Status#TABLE_REMOVE_OPTION_ERROR}.
+     */
+    R<Void> removeOption(String catalogName, String databaseName, String tableName, String key);
+
+    /**
+     * Drops a table from the specified database in the given catalog.
+     *
+     * @param catalogName The name of the catalog from which the table will be dropped.
+     * @param databaseName The name of the database from which the table will be dropped.
+     * @param tableName The name of the table to be dropped.
+     * @return A Response object indicating the success or failure of the operation. If the
+     *     operation is successful, the response will be R.succeed(). If the operation fails, the
+     *     response will be R.failed() with Status.TABLE_DROP_ERROR.
+     * @throws RuntimeException If there is an error during the operation, a RuntimeException is
+     *     thrown with the error message.
+     */
+    R<Void> dropTable(String catalogName, String databaseName, String tableName);
+
+    /**
+     * Renames a table in the specified database of the given catalog.
+     *
+     * @param catalogName The name of the catalog where the table resides.
+     * @param databaseName The name of the database where the table resides.
+     * @param fromTableName The current name of the table to be renamed.
+     * @param toTableName The new name for the table.
+     * @return A Response object indicating the success or failure of the operation. If the
+     *     operation is successful, the response will be R.succeed(). If the operation fails, the
+     *     response will be R.failed() with Status.TABLE_RENAME_ERROR.
+     * @throws RuntimeException If there is an error during the operation, a RuntimeException is
+     *     thrown with the error message.
+     */
+    R<Void> renameTable(
+            String catalogName, String databaseName, String fromTableName, String toTableName);
+
+    /**
+     * Handler method for the "/getAllTables" endpoint. Retrieves information about all tables and
+     * returns a response containing the table details.
+     *
+     * @return Response object containing a list of {@link TableDTO} representing the tables.
+     */
+    R<List<TableDTO>> getAllTables();
+
+    /**
+     * Query tables by catalog id and database name.
+     *
+     * @return Response object containing a list of {@link TableVO} representing the tables.
+     */
+    R<List<TableVO>> getTablesByCondition(Integer catalogId, String databaseName);
```

Comment:
Any difference between this method and `queryTablesByCondition`?

Reply:
queryTablesByCondition is used to search table by name fuzzy. getTablesByCondition is used to query all tables under a database under a catalog

Reply:
@hunter-cloud09, you could merge the two methods and use table object as parameter.